app/services/allocation/AssessorAllocationService.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.allocation
+
+import model.Exceptions.OptimisticLockException
+import model.exchange
+import model.persisted
+import model.command
+import repositories.AssessorAllocationMongoRepository
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object AssessorAllocationService extends AssessorAllocationService {
+  def allocationRepo = repositories.assessorAllocationRepository
+}
+
+trait AssessorAllocationService {
+
+  def allocationRepo: AssessorAllocationMongoRepository
+
+  def getAllocations(eventId: String): Future[command.AssessorAllocations] = {
+    allocationRepo.allocationsForEvent(eventId).map { a => command.AssessorAllocations.apply(a) }
+  }
+
+  def allocate(newAllocations: command.AssessorAllocations): Future[Unit] = {
+    getAllocations(newAllocations.eventId).flatMap { existingAllocation =>
+      existingAllocation.allocations match {
+        case Nil => allocationRepo.save(persisted.AssessorAllocation.fromCommand(newAllocations)).map(_ => ())
+        case _ => updateExistingAllocations(existingAllocation, newAllocations).map(_ => ())
+
+      }
+    }
+  }
+
+  private def updateExistingAllocations(existingAllocation: command.AssessorAllocations,
+    newAllocations: command.AssessorAllocations): Future[Unit] = {
+
+    if (newAllocations.version == existingAllocation.version) {
```

Comment:
So, when we save allocation we need update version (so optimistic lock functionality works as expected).
I assume thats done, I just can't understand/find where it is...
Also, as for me it would be easier to have `version: Int`, and when you need to update it you just increment this number (probably there are valid reason of using String UUID).